docs/topics/spider-middleware.rst

```diff
@@ -98,20 +98,25 @@ object gives you access, for example, to the :ref:`settings <topics-settings>`.
 
     .. method:: process_spider_output(response, result, spider)
 
+        .. versionchanged:: VERSION
+           Since VERSION this can take and return an :term:`python:asynchronous
+           iterable`.
+
         This method is called with the results returned from the Spider, after
         it has processed the response.
 
-        :meth:`process_spider_output` must return an iterable of
-        :class:`~scrapy.http.Request` objects and :ref:`item object
-        <topics-items>`.
+        :meth:`process_spider_output` must return an iterable (normal or
+        asynchronous) of :class:`~scrapy.http.Request` objects and 
+        :ref:`item objects <topics-items>`.
```

Comment:
I think we may need to edit this further to make sure that we do not mislead against what the new note in `coroutines.rst` says.

Maybe a similar note?

```
        :meth:`process_spider_output` must return an iterable of
        :class:`~scrapy.http.Request` objects and 
        :ref:`item objects <topics-items>`.
        
        .. note:: When defined as a :ref:`coroutine <async>`, this method needs 
                  to be an async generator, not just return an iterable.
```

Given a clarification like this, I think we can leave the text below for `:type result:` as is.